bazel/test_external/hello-world.cc

```diff
@@ -1,6 +1,24 @@
+#ifdef __EMSCRIPTEN__
+#include <emscripten.h>
+#include <emscripten/bind.h>
+using namespace emscripten;
+#endif
+
 #include <iostream>
 
+void sayHello() {
+  std::cout << "hello" << std::endl;
+}
+
+#ifdef __EMSCRIPTEN__
+EMSCRIPTEN_BINDINGS(aa) {
+  function("sayHello", &sayHello);
+}
+#else
 int main(int argc, char** argv) {
-  std::cout << "hello world!" << std::endl;
+  std::cout << "start main function" << std::endl;
+  sayHello();
+  std::cout << "end main function" << std::endl;
```

Comment:
Any not include a main function in the program in both cases?   This change seem to turn this example from the executable into a library.  Any particular reason for doing that?

Don't we actually run the result of building this?  (under node?)   We probably should.

Reply:
Both are fine. The example illustrates that we can use `EMSCRIPTEN_BINDINGS` optionally.

Reply:
I updated it so it looks more like what you are looking for.